src/main/kotlin/com/liftric/code/artifact/repository/CodeArtifactRepositoryPlugin.kt

```diff
@@ -53,14 +53,14 @@ fun RepositoryHandler.codeArtifact(domain: String, repository: String): MavenArt
  */
 fun RepositoryHandler.codeArtifact(additionalName: String, domain: String, repository: String) = maven {
     val extensionName = "$additionalName${CodeArtifactRepositoryPlugin.extensionName}"
-    CodeArtifactRepositoryExtension.additional[extensionName]?.let {
+    CodeArtifactRepositoryExtension.additional[additionalName]?.let {
```

Comment:
This ignores the top level extension

Reply:
Ah yes, the `CodeArtifactRepositoryExtension.additional`  workaround isn't identical to the extensions. But if `additionalName` is empty, we won't find the root extension, which isn't part of the `CodeArtifactRepositoryExtension.additional` map

Reply:
With
```
Index: src/main/kotlin/com/liftric/code/artifact/repository/CodeArtifactRepositoryPlugin.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/kotlin/com/liftric/code/artifact/repository/CodeArtifactRepositoryPlugin.kt b/src/main/kotlin/com/liftric/code/artifact/repository/CodeArtifactRepositoryPlugin.kt
--- a/src/main/kotlin/com/liftric/code/artifact/repository/CodeArtifactRepositoryPlugin.kt	(revision 4f8c3a12c4fcc72407dbad0f6d79992083794920)
+++ b/src/main/kotlin/com/liftric/code/artifact/repository/CodeArtifactRepositoryPlugin.kt	(date 1657081258046)
@@ -17,12 +17,16 @@
         when (scope) {
             is Settings -> {
                 scope.extensions.create(extensionName, CodeArtifactRepositoryExtension::class.java, scope.extensions)
+                    .also {
+                        CodeArtifactRepositoryExtension.additional[""] = it
+                    }
             }
-
             is Project -> {
                 scope.extensions.create(extensionName, CodeArtifactRepositoryExtension::class.java, scope.extensions)
+                    .also {
+                        CodeArtifactRepositoryExtension.additional[""] = it
+                    }
             }
-
             else -> {
                 throw GradleException("Should only get applied on Settings or Project")
             }
@@ -81,8 +85,7 @@
  * the default extension
  */
 fun codeArtifactToken(additionalName: String, domain: String): String {
-    val extensionName = "$additionalName${CodeArtifactRepositoryPlugin.extensionName}"
-    val settings = CodeArtifactRepositoryExtension.additional[extensionName]
+    val settings = CodeArtifactRepositoryExtension.additional[additionalName]
         ?: throw GradleException("didn't find CodeArtifactRepositoryExtension with the name: $")
     return settings.authorizationTokenResponse(domain).authorizationToken()
 }
@@ -94,8 +97,7 @@
  * the default extension
  */
 fun codeArtifactUri(additionalName: String, domain: String, repository: String, format: String): URI {
-    val extensionName = "$additionalName${CodeArtifactRepositoryPlugin.extensionName}"
-    val settings = CodeArtifactRepositoryExtension.additional[extensionName]
+    val settings = CodeArtifactRepositoryExtension.additional[additionalName]
         ?: throw GradleException("didn't find CodeArtifactRepositoryExtension with the name: $")
     return settings.repositoryEndpointResponse(domain, repository, format).repositoryEndpoint().let { URI.create(it) }
 }

```
it should work tho?